lib/util/loading_button.dart

```diff
@@ -52,7 +52,11 @@ class _LoadingButtonState extends State<LoadingButton> {
       onPressed: widget.onPressed,
       state: widget.state,
       textStyle: TextStyle(
-        color: Theme.of(context).colorScheme.onPrimary,
+        color: widget.state == ButtonState.fail
```

Comment:
switch statement?

Reply:
Would that be better? You can only execute statements in switch statements, so I'd have to move that into a method or assign the color before the huge return statement... IMO this is clear enough, but I'll leave that up to you

Reply:
Yeah okay I agree. But maybe move the functionality into a method for readability